pkg/provider/kubernetes/ingress/kubernetes.go

```diff
@@ -529,8 +529,8 @@ func loadService(client Client, namespace string, backend networkingv1beta1.Ingr
 		return nil, errors.New("subset not found")
 	}
 
-	var port int32
 	for _, subset := range endpoints.Subsets {
+		var port int32
```

Comment:
I guess that this was a mere oversight, by declaring the variable before the cycle its value remained unchanged in subsequent cycles thus (incorrectly) falling through the `if port == 0` guard.